pkg/cluster/connection_pooler.go

```diff
@@ -1037,10 +1037,47 @@ func (c *Cluster) syncConnectionPoolerWorker(oldSpec, newSpec *acidv1.Postgresql
 			syncReason = append(syncReason, specReason...)
 		}
 
+		listOptions := metav1.ListOptions{
+			LabelSelector: labels.Set(c.connectionPoolerLabels(role, true).MatchLabels).String(),
+		}
+		pods, err = c.listPoolerPods(listOptions)
 		newPodAnnotations := c.annotationsSet(c.generatePodAnnotations(&c.Spec))
 		if changed, reason := c.compareAnnotations(deployment.Spec.Template.Annotations, newPodAnnotations); changed {
 			specSync = true
 			syncReason = append(syncReason, []string{"new connection pooler's pod template annotations do not match the current ones: " + reason}...)
+
+			if strings.Contains(reason, "Removed") {
+				annotationToRemove := `{"metadata":{"annotations":{`
+				annotationToRemoveTemplate := `{"spec":{"template":{"metadata":{"annotations":{`
+				for anno := range deployment.Spec.Template.Annotations {
+					if _, ok := newPodAnnotations[anno]; !ok {
+						// template annotation was removed
+						for _, ignore := range c.OpConfig.IgnoredAnnotations {
+							if anno == ignore {
+								continue
+							}
+						}
+						annotationToRemove += fmt.Sprintf(`"%s":null,`, anno)
+						annotationToRemoveTemplate += fmt.Sprintf(`"%s":null,`, anno)
+					}
+				}
+				annotationToRemove = strings.TrimSuffix(annotationToRemove, ",") + `}}}`
+				annotationToRemoveTemplate = strings.TrimSuffix(annotationToRemoveTemplate, ",") + `}}}}}`
```

Comment:
can we not use json marshalling here? This concatenating {{{ + }}} looks a bit "ugly".